waffle/utils.py

```diff
@@ -1,9 +1,11 @@
 from __future__ import unicode_literals, absolute_import
 
 import hashlib
+import ipaddress
 
 from django.conf import settings
 from django.core.cache import caches
+from ipware import get_client_ip
```

Comment:
I'd love to see CIDR ranges supported. I've dealt with client IP a lot in django-ratelimit as well, and I'd consider possibly moving this to something like `waffle.ext....`

As ipware's readme indicates, correctly determining client IP behind a proxy is _super_ environment-specific (this is why Django itself dropped the `SetRemoteAddrFromForwardedFor` middleware [all the way back in 1.1](https://docs.djangoproject.com/en/1.8/releases/1.1/#removed-setremoteaddrfromforwardedfor-middleware)). I'd be strongly tempted to defer that to application code (or another library if you want to configure ipware).

Maybe make the protocol something like: set a value on the request (possibly `request.META['WAFFLE_IP']` or something) and Waffle will use that value, falling back to `request.META['REMOTE_ADDR']`.

Then you could provide middleware e.g. `waffle.ext.middleware.WaffleClientIPMiddleware` that depends on `ipware` (but only add it as an optional dependency, and add a system check to the middleware module for it) and sets `request.META['WAFFLE_IP']`.

That way users could also optionally write their own middleware to set the value based on their environments. For example, on my own projects I usually like to use something like `X-Client-IP/Proto` or `Cluster-Client-IP/Proto` from the edge, to avoid dealing with parsing XFF correctly, so I could write a small middleware to copy `HTTP_X_CLIENT_IP` to `WAFFLE_IP`.

It's also probably worth scoping these fields to IPv4 to enable expansion to IPv6 ranges in the future